isaactest/tests/board_builder.py

```diff
@@ -0,0 +1,163 @@
+    for question_result in question_results:
+        assert assertion(question_result), failure_message_formatter(question_result.text)
+
+
+@TestWithDependency('BOARD_BUILDER')
+def board_builder(driver, Users, ISAAC_WEB, WAIT_DUR, **kwargs):
+    try:
+        random_id = ''.join(random.choice('ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890') for _ in range(8))
+
+        log(INFO, "Log in as user and go to page")
+        assert_tab(driver, ISAAC_WEB)
+        driver.get(ISAAC_WEB + '/logout')
+        driver.get(ISAAC_WEB + '/game_builder')
+        assert submit_login_form(driver, user=Users.Admin, wait_dur=WAIT_DUR), "Can't access User Admin; can't continue testing!"
+
+        log(INFO, "Test subject field filtering")
+        subject_field = Select(driver.find_element_by_xpath('//select[@ng-model="questionSearchSubject"]'))
+
+        subject_field.select_by_visible_text('Mathematics')
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: 'maths' in question_result.text.lower(),
+                          lambda question_result_text: '"maths" was not found in "{}"'.format(question_result_text.lower()))
+
+        subject_field.select_by_visible_text('Physics')
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: 'physics' in question_result.text.lower(),
+                          lambda question_result_text: '"physics" was not found in "{}"'.format(question_result_text.lower()))
+
+        log(INFO, "Test level field filtering")
+        level_field = Select(driver.find_element_by_xpath('//select[@ng-model="questionSearchLevel"]'))
+        level_field.select_by_visible_text('2')
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: 'level 2' in question_result.text.lower(),
+                          lambda question_result_text: '"level 2" was not found in "{}"'.format(question_result_text.lower()))
+
+        log(INFO, "Test query field filtering")
+        query_field = driver.find_element_by_xpath('//input[@ng-model="questionSearchText"]')
+        query_field.send_keys('toboggan')
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: 'toboggan' in question_result.text.lower(),
+                          lambda question_result_text: '"toboggan" was not found in "{}"'.format(question_result_text.lower()))
+
+        log(INFO, "Check book links work")
+        mastering_physics_link = driver.find_element_by_xpath('//a[text() = "Mastering Physics"]')
+        mastering_physics_link.click()
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: all(keyword in question_result.text.lower() for keyword in ['book', 'physics', 'physics_skills_14']),
```

Comment:
The "book" tag is likely to be deprecated soon, so just `physics_skills_14` should be enough.